### DISCUSSION TITLE
How to nest atomfamily of atomfamilies ?

### DISCUSSION THREAD

**Discussion by GuacheSuede** [facebookexperimental/Recoil#1965] [Q&A]
How would one atomize complex structures say 3-4 levels deep ? Would it be possible to have nested atom families of families ? Thanks.

EG.

```

video = [
    frames = [
         frame = [
             info: {type: '',  date: ''}
             data: ''
         ]
    subtitles: .....etc
    ]
]
```

**Comment by serhii-babich** [comment]
Try to «normalize» your state by storing all entities in separate atomFamilies, keepin only Id references in parent structures and then selecting them by Ids. Like this:

```ts
const frames$ = atomFamily({
  key: "@frame",
  default: null,
});

interface Video {
  frameIds: string[];
}
const videos$ = atomFamily({
  key: "@video",
  default: null,
});

const framesByVideo = selectorFamily$({
  key: "@FramesByVideo",
  get(videoId: string) {
    return ({ get }) => {
      const video = get(videos$(videoId));
      const frames = video.frameIds.map((frameId) => get(frames$(frameId)));
      return frames;
    };
  },
});

```

**Comment by GuacheSuede** [comment]
Thank U for the detailed answer !

**Comment by GuacheSuede** [comment]
How would one completely remove an atom from a atom family ?

**Comment by serhii-babich** [comment]
@GuacheSuede by resetting it to default value or by setting it to undefined.

**Comment by GuacheSuede** [comment]
@sergiybabich Is a way to remove/filter it to reduce copy operations ?

**Comment by serhii-babich** [comment]
@GuacheSuede what exactly do you mean?

**Comment by GuacheSuede** [comment]
@sergiybabich Is it possible to remove a atom from a existing atom family without needed to reset it first ?
